mixer/cmd/root.go

```diff
@@ -75,8 +75,17 @@ var RootCmd = &cobra.Command{
 			fail(err)
 		}
 
+		networkCheck := true
+		noNetworkCmds := []string{"list", "edit", "validate", "convert", "set", "repo", "add-rpms"}
+		// Don't reach out over network for these commands, it's not needed
+		for _, ignoreCmd := range noNetworkCmds {
+			if cmdContains(cmd, ignoreCmd) {
+				networkCheck = false
+			}
+		}
+
 		// If running natively, check for format missmatch and warn
-		if builder.Native && b.UpstreamURL != "" {
+		if networkCheck && builder.Native && b.UpstreamURL != "" {
 			hostFormat, upstreamFormat, err := b.GetHostAndUpstreamFormats()
```

Comment:
Looking at the code below, it also seems weird to me that this should fail() on error instead of just printing the first warning. Afteral, when you get an error here you were "Unable to determine host format".

Reply:
Hmm, you make a good point. I might have to update this but I want to think about it more :)